pkg/controllers/nodeclass/controller.go

```diff
@@ -172,6 +226,16 @@ func (c *NodeClassController) Builder(_ context.Context, m manager.Manager) core
 	return corecontroller.Adapt(controllerruntime.
 		NewControllerManagedBy(m).
 		For(&v1beta1.EC2NodeClass{}).
+		Watches(
+			&source.Kind{Type: &corev1beta1.NodeClaim{}},
+			handler.EnqueueRequestsFromMapFunc(func(o client.Object) []reconcile.Request {
+				nc := o.(*corev1beta1.NodeClaim)
+				if nc.Spec.NodeClass == nil {
+					return nil
+				}
+				return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: nc.Spec.NodeClass.Name}}}
+			}),
```

Comment:
We only need to watch NodeClaims delete events, right? Or is it update because the finalizer would be removed?

Reply:
You're right. I can constrain this to deletes